File: Domain/Entites/WarehouseItem.cs
using Application.ViewModels;
using Domain.Common;

namespace Domain.Entites
{
    public class WarehouseItem : AuditableEntity
    {
        public string Location { get; set; }
        public int? ProductID { get; set; }
        public int? BatchID { get; set; }
        public int? WarehouseId { get; set; }
        public int? OrderDetailId { get; set; }
        public virtual Product Product { get; set; }
        public virtual Warehouse Warehouse { get; set; }
        public virtual Allocation Allocation { get; set; }
    }
}

File: Managerial/Managerial/ClientApp/src/app/models/warehouse/WarehouseItem.ts
import { Order } from "../order/Order.model";
import { Batch } from "../product/Batch";
import { Product } from "../product/Product.model";
import { Allocation } from "./Allocation";
import { Warehouse } from "./Warehouse.model";

export class WarehouseItem {


  public Product: Product
  public Warehouse: Warehouse
  public Allocation: Allocation
  public Location: Location
  public Batch: Batch
}
